### PR TITLE
[Serverless] Twin PR, AI use cases

### LINKED ISSUE #5456: [Request] Gen AI section, use-cases docs, parity
### Description

Bring the serverless GenAI docs section to parity with the ESS version. Some of the changes currently included in the ESS docs are not yet present in serverless, but have been approved for inclusion in both docs sets. 

### Background & resources

* PRs: 
* Issues/metas: 
* Point of contact: 
* Test environments: 

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.15

### Serverless release

-

### Feature differences

-

### API docs impact

-

### Prerequisites, privileges, feature flags

_No response_

### PR DESCRIPTION
**Pull request by benironside**
Fixes #5456 by implementing some Gen AI docs changes in serverless that are already present in ESS. These changes are already present in ESS, so a thorough review is not necessary.

Preview: [AI use cases section](https://elastic-dot-co-docs-production-p4i1lot7x-elastic-dev.vercel.app/current/serverless/security/llm-connector-guides)

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-f83fsd32p-elastic-dev.vercel.app

* Built with commit de4472957206a5227408e9541a4bf72b83a80105

> Issues? Visit #next-docs in Slack

File: docs/serverless/assistant/llm-connector-guides.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+slug: /serverless/security/llm-connector-guides
+title: LLM connector guides
+description: Set up LLM connectors to enable AI features in ((elastic-sec))
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/llm-connector-guides.mdx:4]:
```suggestion
description: Set up LLM connectors to enable AI features in ((elastic-sec)).
```

File: docs/serverless/assistant/usecase-attack-disc-ai-assistant-incident-reporting.mdx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+slug: /serverless/security/ai-usecase-incident-reporting
+title: Identify, investigate, and document threats
+description: Use Attack discovery and AI Assistant to manage threats
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/usecase-attack-disc-ai-assistant-incident-reporting.mdx:4]:
```suggestion
description: Use Attack discovery and AI Assistant to manage threats.
```

File: docs/serverless/assistant/ai-assistant-esql-queries.mdx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+slug: /serverless/security/ai-assistant-esql-queries
+title: Generate, customize, and learn about ((esql)) queries
+description: AI Assistant has specialized ((esql)) capabilities
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/ai-assistant-esql-queries.mdx:4]:
```suggestion
description: AI Assistant has specialized ((esql)) capabilities.
```

**Review by nastasha-solomon — APPROVED** [review]

**Pull request merged**